commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java

```diff
@@ -296,4 +309,17 @@ private int executeCommand(final String command, final StringBuilder output) thr
         channel.disconnect();
         return channel.getExitStatus();
     }
+
+    /**
+     * Some SFTP-only servers disable the exec channel.
+     * 
+     * Attempt to detect this by calling getUid.
+     */
+    private void detectExecDisabled() {
+        try {
+            getUId();
```

Comment:
Should this value be cached?

Reply:
The UID? It probably could be, I can't imagine it changing at runtime. But it wasn't cached and I figured I'd keep my patch as small as possible.